### PR TITLE
fix(curriculum): Shortens Date Formatter Project

### LINKED ISSUE #55331: Steps 23, 24, and 25 of the date formatter project can be combined
### Describe the Issue

In the steps 23, 24, and 25 of the date formatter, campers are asked campers to use the `split()`, `reverse()`, and `join()` methods on a `formattedDate` variable.

I think these steps should be combined, since steps 18 through 22 detailed how to use those methods and chain them.

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-the-date-object-by-building-a-date-formatter/step-23, https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-the-date-object-by-building-a-date-formatter/step-24, https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-the-date-object-by-building-a-date-formatter/step-25

### Your code

N/A

### Expected behavior

The steps should be combined

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by patrickwcode**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #55331

- Combines Steps 23 to 25 for the Date Formatter project in English.
- Removes Steps 23 and 24.
- Total Steps updated from 29 to 27.
- Lesson, hint, and code are updated and working.

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-the-date-object-by-building-a-date-formatter/6538935e2ab721beedb137c4.md [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 
 Test out your changes by selecting the `Year, Month, Day` option from the dropdown menu. The date should now be displayed in the format `yyyy-mm-dd`.
 
 # --hints--
 
-You should use the `.join()` method to join the reversed array elements into a string and use a `"-"` as a separator.
+The `.join()` and `.split()` methods should use a `"-"` as a separator.  Make sure to use `.reverse()` at the end.
 
 ```js
 assert(code.match(/\.join\(\s*('|")-\1\s*\)/g));
```
Comment (moT01) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-the-date-object-by-building-a-date-formatter/6538935e2ab721beedb137c4.md:19]: This test should be updated to check that all three things are used. Or more tests should be added to check them individually.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-the-date-object-by-building-a-date-formatter/6538935e2ab721beedb137c4.md [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
+title: Step 23
 challengeType: 0
-dashedName: step-25
+dashedName: step-23
 ---
 
 # --description--
 
-Finally, join the results from the array reversal. Use a `"-"` as the separator. Remember to use method chaining.
+Like in the previous step, use method chaining to `split`, `reverse`, and `join` the array.
```
Comment (moT01) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-the-date-object-by-building-a-date-formatter/6538935e2ab721beedb137c4.md:10]:
```suggestion
Like in the previous step, use method chaining to `split`, `reverse`, and `join` the array. Use `'-'` in the split and join methods.
```

I think we should add a mention here to use the `-` on the split and join.

**Review by moT01 — COMMENTED** [review]
Looking good @patrickwcode 👍 I left a few suggestions. Let us know if you need help with the test.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-the-date-object-by-building-a-date-formatter/6538935e2ab721beedb137c4.md (hunk 2 above)
Comment (patrickwcode) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-the-date-object-by-building-a-date-formatter/6538935e2ab721beedb137c4.md:10]: Thank you very much! Would it be okay if I look into this next week? I will let you know if I need any help.

**Comment by moT01** [comment]
There's a conflict here @patrickwcode, I think updating your branch will take care of it. Let me know if you want me to take a look.

**Comment by patrickwcode** [comment]
@moT01 My apologies.  I am reading the freeCodeCamp docs for guidance.  Should I take these steps below?

1. Rebase my copy
2. Remove the conflicting files (`curriculum/i18n-curriculum`)
3. Push changes to PR

**Comment by ilenia-magoni** [comment]
Will you continue working on this, @patrickwcode ? if you need help you should join our [Discord](https://discord.gg/KVUmVXA) server and ask for help in the `#contributors` channel. That will help other contributors guide you to get your PR ready for a review.

**Comment by moT01** [comment]
I went ahead and fixes the issues here. It should be good to go.

**Comment by ilenia-magoni** [comment]
@moT01 the tests still fail, step 23 has something wrong with it

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Review by moT01 — APPROVED** [review]

**Pull request merged**